src/generators/ruby/index.js

```diff
@@ -0,0 +1,450 @@
+  setupHandlebars() {
+    let that = this;
+    Handlebars.registerHelper("pascalCasePropName", function() {
+      return new Handlebars.SafeString(that.convertToCamelCase(this.propName));
+    });
+
+    Handlebars.registerHelper("snakePropName", propName => {
+      return new Handlebars.SafeString(that.canonicalToSnakeName(propName));
+    });
+
+    Handlebars.registerHelper("module_start", context => {
+      return "  ".repeat(context);
+    });
+    Handlebars.registerHelper("module_end", function(context, options) {
+      let modules = options.data.root.namespace_parts.length;
+
+      return "  ".repeat(modules - context - 1);
+    });
+    Handlebars.registerHelper("indent", function(context, options) {
+      let modules = context.data.root.namespace_parts.length;
+      let indent = "  ".repeat(modules);
+
+      let content = context.fn(this);
+
+      let formatted =
+        content
+          .trimEnd()
+          .split("\n")
+          .map(line => {
+            if (line.trimEnd().length > 0) {
+              return `${indent}${line}`;
+            } else {
+              return "";
+            }
+          })
+          .join("\n") + "\n";
+
+      return formatted;
+    });
+  }
+
+  async renderModel(data) {
+    data["namespace_parts"] = this.getNamespaceParts(data.modelType, "Models");
+
+    this.modelTemplate =
+      this.modelTemplate ||
+      (await this.loadTemplate(__dirname + "/model.rb.mustache"));
+
+    return this.modelTemplate(data);
+  }
+
+  async renderEnum(data) {
+    const includedInSchema = this.includedInSchema(data.enumType);
+
+    data["namespace_parts"] = this.getNamespaceParts(data.enumType, "Enums");
+
+    this.enumTemplate = this.enumTemplate || {
+      main: await this.loadTemplate(__dirname + "/enum_main.rb.mustache")
+    };
+
+    let response = {
+      [this.getEnumFilename(data)]: this.enumTemplate.main(data)
+    };
+
+    return response;
+  }
+
+  createIndexFiles() {
+    return {
+      "/files_index.json": JSON.stringify(this.generatedFiles, null, 2)
+    };
+  }
+
+  getDirs() {
+    return ["models/", "models/schema/", "enums/"];
+  }
+
+  getModelFilename(model) {
+    let parts = [
+      "models",
+      ...this.getBasicNamespace(model.type),
+      this.getPropNameFromFQP(model.type)
+    ];
+
+    parts = parts
+      .filter(val => !!val)
+      .map(name => {
+        console.log(name);
+        return this.canonicalToSnakeName(name);
+      });
+
+    return "/" + parts.join("/") + ".rb";
+  }
+
+  getEnumFilename(thisEnum) {
+    let parts = [
+      "enums",
+      ...this.getBasicNamespace(thisEnum.enumType),
+      this.getPropNameFromFQP(thisEnum.enumType)
+    ];
+
+    parts = parts
+      .filter(val => !!val)
+      .map(name => {
+        return this.canonicalToSnakeName(name);
+      });
+
+    return "/" + parts.join("/") + ".rb";
+  }
+
+  convertToClassName(value) {
+    // 3DModel is an invalid class name..
+    value = value.replace(/^3/, "Three");
```

Comment:
Just noticed this - good point, and it's the same for PHP. Gonna whip a quick PR for it to get merged into yours.